src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -670,7 +653,61 @@ private static FunctionCompletedMessage CreateCompletedMessage(FunctionStartedMe
                 ParameterLogBlob = startedMessage.ParameterLogBlob
             };
         }
-                
+
+        // Called very early when function is started; before arguments are bound. 
+        private Task NotifyPreBindAsync(FunctionInstanceLogEntry fastItem)
+        {
+            fastItem.LiveTimer = Stopwatch.StartNew();
+
+            if (_functionEventCollector == null)
+            {
+                return Task.CompletedTask;
+            }            
+
+            // Log pre-bind event. 
+            return _functionEventCollector.AddAsync(fastItem);
+        }
+
+        // Called before function body is executed; after arguments are bound. 
+        private Task NotifyPostBindAsync(FunctionInstanceLogEntry fastItem, IDictionary<string, string> arguments)
```

Comment:
this was intentional - I wanted the sigs here to enforce some basic invariants; like requiring an arguments was passed in. 

---
In reply to: [126782972](https://github.com/Azure/azure-webjobs-sdk/pull/1218#discussion_r126782972) [](ancestors = 126782972)